### ISSUE TITLE
"#warning "crt/link.stub is an internal header" error when compiling TensorFlow w/ CUDA 10

### ISSUE THREAD

**Issue by reedwm** [tensorflow/tensorflow#22925]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 53faa313b7628cd8c9fbb836544cc6482cafb7a4
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.15.2
- **GCC/Compiler version (if compiling from source)**: 7.3.0
- **CUDA/cuDNN version**: CUDA 10.0, cudnn 7.3
- **GPU model and memory**: V100
- **Exact command to reproduce**:

```
yes '' | TF_NEED_CUDA=1 TF_CUDA_VERSION=10.0 CUDNN_INSTALL_PATH=/usr/local/cudnn NCCL_INSTALL_PATH=/usr/local/nccl_2.3.4-1+cuda10.0_x86_64  ./configure
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package
```

### Describe the problem
Starting at commit 53faa313b7628cd8c9fbb836544cc6482cafb7a4, when trying to compile TensorFlow with CUDA 10.0, I get an error stating "#warning "crt/link.stub is an internal header file and must not be used directly". The full error output is:

```
ERROR: /home/reedwm/.cache/bazel/_bazel_reedwm/f53335a0bee8ade861f80a7955d79beb/external/nccl_archive/BUILD.bazel:139:1: C++ compilation of rule '@nccl_archive//:device_code' failed (Exit 1)
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc:19:2: warning: #warning "crt/link.stub is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead." [-Wcpp]
 #warning "crt/link.stub is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead."
  ^~~~~~~
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc: In function 'void __cudaRegisterLinkedBinary(const __fatBinC_Wrapper_t*, void (*)(void**), void*)':
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc:140:5: error: '__NV_EXTRA_INITIALIZATION' was not declared in this scope
     __NV_EXTRA_INITIALIZATION
     ^~~~~~~~~~~~~~~~~~~~~~~~~
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc:144:5: error: '__NV_EXTRA_FINALIZATION' was not declared in this scope
     __NV_EXTRA_FINALIZATION
     ^~~~~~~~~~~~~~~~~~~~~~~
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc:145:23: warning: statement has no effect [-Wunused-value]
     for (__i = 0; __i < NUM_PRELINKED_OBJECTS; ++__i) {
bazel-out/k8-opt/bin/external/nccl_archive/device_code.cc:145:53: error: expected ';' before ')' token
     for (__i = 0; __i < NUM_PRELINKED_OBJECTS; ++__i) {
                                                     ^
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.

```
This does not occur before 53faa313b7628cd8c9fbb836544cc6482cafb7a4. Letting TensorFlow download and build NCCL does not resolve this issue.

@chsigg, can you resolve this?

**Comment by richardburleigh** [comment]
Same exact error occurs with Python 3.7 as well.

**Comment by ng0177** [comment]
same here.: Highly appreciate fixing it!

**Comment by chsigg** [comment]
device_code.cc is actually a file from a template that ships with the CUDA SDK (cuda/bin/crt/link.stub). I will prepare a CL to replace that include along with the other substitutions. In the meantime, you can install NCCL and specify the version and install location during ./configure.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 19 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by reedwm** [comment]
This was fixed by 8c072a519e2beed483adb361a9be934a47bee366.
